sdk/cosmos/azure-cosmos-encryption/src/test/java/com/azure/cosmos/encryption/CosmosEncryptionItemSerializerTest.java

```diff
@@ -0,0 +1,810 @@
+        }
+        List<T> results = container
+            .queryItems("SELECT * FROM c where c.id = '" + id + "'", queryRequestOptions, classType)
+            .stream().collect(Collectors.toList());
+        assertThat(results).isNotNull();
+        assertThat(results).hasSize(1);
+        assertSameDocument(doc, results.get(0));
+    }
+
+    @Test(groups = { "fast", "emulator" }, dataProvider = "testConfigs_requestLevelSerializer", timeOut = TIMEOUT * 1000000)
+    public void bulkAndReadManyWithObjectNode(CosmosItemSerializer requestLevelSerializer) {
+
+        runBulkAndReadManyTestCase(
+            id -> TestDocument.createAsObjectNode(id),
+            requestLevelSerializer,
+            ObjectNode.class
+        );
+    }
+
+    @Test(groups = { "fast", "emulator" }, dataProvider = "testConfigs_requestLevelSerializer", timeOut = TIMEOUT * 1000000)
+    public void bulkAndReadManyWithPojo(CosmosItemSerializer requestLevelSerializer) {
+
+        runBulkAndReadManyTestCase(
+            id -> TestDocument.create(id),
+            requestLevelSerializer,
+            TestDocument.class
+        );
+    }
+
+
+    private <T> void runBulkAndReadManyTestCase(
+        Function<String, T> docGenerator,
+        CosmosItemSerializer requestLevelSerializer,
+        Class<T> classType) {
+
+        CosmosBulkExecutionOptions bulkExecOptions = new CosmosBulkExecutionOptions()
+            .setCustomSerializer(requestLevelSerializer);
+
+        List<CosmosItemOperation> bulkOperations = new ArrayList<>();
+        Map<String, T> inputItems = new HashMap<>();
+        for (int i = 0; i < 10; i++) {
+            String id = UUID.randomUUID().toString();
+            T doc = docGenerator.apply(id);
+            inputItems.put(id, doc);
+            bulkOperations.add(CosmosBulkOperations.getCreateItemOperation(
+                doc,
+                new PartitionKey(id),
+                id));
+        }
+
+        Iterable<CosmosBulkOperationResponse<Object>> responseFlux =
+            container.executeBulkOperations(bulkOperations, bulkExecOptions);
+        for(CosmosBulkOperationResponse<Object> response: responseFlux) {
+            assertThat(response.getException()).isNull();
+            assertThat(response.getResponse()).isNotNull();
+            assertThat(response.getResponse().getStatusCode()).isBetween(200, 201);
+            assertThat(response.getOperation().<String>getContext()).isNotNull();
+            String id = response.getOperation().getContext();
+            assertThat(inputItems.containsKey(id)).isTrue();
+            T responseItem = response.getResponse().getItem(classType);
+
+            if (isContentOnWriteEnabled) {
+                assertSameDocument(inputItems.get(id), responseItem);
+            } else {
+                assertThat(responseItem).isNull();
+            }
+        }
+
+        // TODO - re-enable the read many validation after adding readMany API for azure-cosmos-encryption
```

Comment:
should we also create a GitHub working item for tracking it?

Reply:
Done